src/components/forms/LocationSelect/index.tsx

```diff
@@ -1,11 +1,123 @@
-import { FieldProps } from 'formik';
-import React from 'react';
-import JurisdictionSelect, { JurisdictionSelectProps } from '../JurisdictionSelect';
-const LocationSelect = (props: JurisdictionSelectProps & FieldProps) => {
-  const newProps = {
-    ...props,
-    loadLocations: true,
+import React, { useEffect, useState } from 'react';
+import TreeMenu, { ItemComponent, TreeNode } from 'react-simple-tree-menu';
+import 'react-simple-tree-menu/dist/main.css';
+import { toast } from 'react-toastify';
+import { Col, Row } from 'reactstrap';
+import { LOADING, LOWEST_JURISDICTION } from '../../../configs/lang';
+import { OPENSRP_ACTIVE } from '../../../constants';
+import { displayError } from '../../../helpers/errors';
+import { growl } from '../../../helpers/utils';
+import { getFilterParams, OpenSRPService } from '../../../services/opensrp';
+import { NoDataComponent } from '../../Table/NoDataComponent';
+import { FIND_BY_PROPERTIES } from '../../TreeWalker/constants';
+import { JurisdictionOption } from '../JurisdictionSelect';
+import { buildLocationTree, JurisdictionTreeById } from './helpers';
+import './index.css';
+import StudentExportForm from './StudentExportForm';
+
+interface LocationSelectProps {
+  endpoint: string;
+  service: typeof OpenSRPService;
+}
+
+const LocationSelect = (props: LocationSelectProps) => {
+  const { service, endpoint } = props;
+  const [treeData, setTreeData] = useState<JurisdictionTreeById>({});
+  const [loadingId, setLoadingId] = useState<string | null>(null);
+  const [openNodes, setOpenNodes] = useState<string[]>([]);
+
+  const getJurisdiction = (
+    parentId: string = '',
+    key: string = '',
+    clickedNodes: string[] = []
+  ) => {
+    if (loadingId) {
+      return;
```

Comment:
Is this what we want to happen when it is loading?

Reply:
Yes

Reply:
Generally it ensures we load features of one location at a time

Reply:
What I meant is: do we need any indication that something is loading?